coriolis/tests/test_data_transfer.py

```diff
@@ -0,0 +1,152 @@
+        self.offset = 0
+
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address(self, mock_conf):
+        mock_conf.compressor_address = 'localhost:8000'
+        result = data_transfer._get_session_and_address()
+
+        self.assertIsInstance(result[0], requests.Session)
+        self.assertEqual(result[1], "http://%s/" %
+                         mock_conf.compressor_address)
+
+    @mock.patch.object(data_transfer.os, 'stat')
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address_unix_socket(self, mock_conf, mock_stat):
+        mock_conf.compressor_address = '/var/run/compressor.sock'
+        mock_stat.return_value.st_mode = stat.S_IFSOCK
+
+        result = data_transfer._get_session_and_address()
+
+        self.assertIsInstance(result[0], requests_unixsocket.Session)
+        self.assertEqual(result[1], "http+unix://%s/" % parse.quote_plus(
+            mock_conf.compressor_address))
+
+    @mock.patch.object(data_transfer.os, 'stat')
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_get_session_and_address_invalid_unix_socket(self, mock_conf,
+                                                         mock_stat):
+        mock_conf.compressor_address = '/var/run/compressor.sock'
+        mock_stat.return_value.st_mode = stat.S_IFREG
+
+        self.assertRaises(exception.CoriolisException,
+                          data_transfer._get_session_and_address)
+
+    @mock.patch.object(data_transfer.constants, 'COMPRESSION_FORMAT_GZIP')
+    def test_compression_proxy_invalid_format(self, mock_compression_format):
+        self.assertRaises(exception.CoriolisException,
+                          data_transfer.compression_proxy,
+                          mock.sentinel.content,
+                          mock_compression_format.return_value)
+
+    @mock.patch.object(data_transfer, '_COMPRESS_FUNC')
+    def test_compression_proxy(self, mock_compress_func):
+        result = data_transfer.compression_proxy(self.data_content, self.fmt)
+
+        mock_compress_func[self.fmt].assert_called_once_with(self.data_content)
+        self.assertEqual(
+            result, (mock_compress_func[self.fmt].return_value, True))
+
+    @mock.patch.object(data_transfer, '_get_session_and_address')
+    @mock.patch.object(data_transfer, 'CONF')
+    def test_compression_proxy_with_compression(self, mock_conf,
+                                                mock_get_session_and_address):
+        mock_get_session_and_address.return_value = (
+            self.mock_session, self.mock_url)
+        result = data_transfer.compression_proxy(self.data_content, self.fmt)
+
+        self.mock_session.post.assert_called_once_with(
+            self.mock_url, data=self.data_content,
+            headers={'X-Compression-Format': self.fmt},
+            timeout=mock_conf.default_requests_timeout)
+        self.assertEqual(
+            result, (self.mock_session.post.return_value.content, False))
```

Comment:
Additionally, please check that `self.mock_session.close` is called once, thus making sure the session gets closed after the request is sent